Integrations/MimecastV2/MimecastV2.py

```diff
@@ -443,43 +448,169 @@ def create_policy():
+            'Domain': sender.get('emailDomain'),
+            'Type': sender.get('type')
+        },
+        'Reciever': {
+            'Group': receiver.get('groupId'),
+            'Email Address': receiver.get('emailAddress'),
+            'Domain': receiver.get('emailDomain'),
+            'Type': receiver.get('type')
+        },
+        'Bidirectional': policy.get('bidirectional'),
+        'Start': policy.get('fromDate'),
+        'End': policy.get('toDate')
+    }
+    policies_context = {
+        'ID': policy_id,
+        'Description': description,
+        'Sender': {
+            'Group': sender.get('groupId'),
+            'Address': sender.get('emailAddress'),
+            'Domain': sender.get('domain'),
+            'Type': sender.get('type')
+        },
+        'Reciever': {
+            'Group': receiver.get('groupId'),
+            'Address': receiver.get('emailAddress'),
+            'Domain': receiver.get('emailDomain'),
+            'Type': receiver.get('type')
+        },
+        'Bidirectional': policy.get('bidirectional'),
+        'FromDate': policy.get('fromDate'),
+        'ToDate': policy.get('toDate')
+    }
+
+    context['Mimecast.Policy(val.ID && val.ID == obj.ID)'] = policies_context
+
+    results = {
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': raw_response,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown(title, raw_response, headers),
+        'EntryContext': context
+    }
+
+    return results
+
+
+def create_policy_request(policy, option):
+    # Setup required variables
+    api_endpoint = '/api/policy/blockedsenders/create-policy'
+    payload = {
+        'data': [{
+            'policy': policy,
+            'option': option
+        }]
+    }
+    response = http_request('POST', api_endpoint, str(payload))
+    if response.get('fail'):
+        return_error(json.dumps(response.get('fail')[0].get('errors')))
+    return response.get('data')[0]
+
+
+def set_empty_value_args(policy_obj, option, policy_id):
+    """
+      Args:
+        policy_obj (Dict): Dict of policy details
+        option: (str) Policy option
+        policy_id: (str) Policy ID
+
+      Returns:
+          Returning a tuple with all of policy arguments filled with data (Tuple).
+     """
+    empty_args_list = []
+    for arg, value in policy_obj.items():
+        if value == '':
+            empty_args_list.append(arg)
+    if option == '':
+        empty_args_list.append("option")
+    if len(empty_args_list) > 0:
+        policy_details = get_policy_request(policy_id)
+        for arg in empty_args_list:
+            if arg == "option":
+                option = policy_details[0]["option"].encode("utf-8")
+            else:
+                policy_obj[arg] = policy_details[0]["policy"][arg].encode("utf-8")
+
+    return policy_obj, option, policy_id
+
+
+def update_policy(args):
```

Comment:
this function is too long at the moment.
removing duplicate code and extracting logic to external functions could help

Reply:
Rony told me to leave the content and context the way they are right now